my_class.h

```diff
@@ -0,0 +1,39 @@
+//@file: my_class.h
+//@author: Veselina Velkova
+//@brief: This file contains the prototypes of the functions in the student class.
+//@date: 18.04.2023
+
+#pragma once
+#include<string>
+
+//These constants are useful because in real life students can not have infinite number of grades.
+//Usually, for each subject there is minimum and maximum number of grades that each student should have. 
+const int MIN_GRADES = 5;
+const int MAX_GRADES = 20;
+
+class student
+{
+    std::string name;
+    int number;
+    int numberGrades;
+    int currentCapacity = 5;
+    float sum = 0.0;
```

Comment:
I'm new to c++.. stupid qn..
Is it acceptable to provide initial values for parameters within the class directly?
In C, you cannot do that. I had to have an init function to set up initial values/malloc/calloc and then free functions.
What's the convention with regards to init values like this?
What happens when this class is being inherited by another and maybe that new class has different init values? Does the new class just overwrite them?

Reply:
About this question, it is possible for sure - I think the difference is that like this the default values are added immediately while constructing. The other way with the set functions first uses the constructor, the attributes still don't have any value and then we add the default value (or in other words would be slowlier for bigger types). I just read this here: https://stackoverflow.com/questions/7207884/class-construction-with-initial-values - haven't thought about this until now. As for the inheritance I am not really sure...

Reply:
OK.. thank you for the link... now, follow up qn.
Which one is "more" correct?
class Student
{
float sum = 0.0;
Student(): { /* empty */ }
}

VS.

class Student
{
float sum;
Student(): sum(0.0){...}
}

Reply:
I think they are both fine. The first option - when the constructor is empty, there is no need to write it (there is a default constructor by the compiler) https://www.geeksforgeeks.org/default-constructors-in-cpp/ If I have understood correctly the previous link - there won't be a huge difference in the time, because float is a simple type. Just a little fact that I noticed when I wrote my class - at the beginning I put the dynamic array allocation in the constructor. But then all the other methods which used it gave me error that they don't know what grades is. And that's the reason why I put it outside. With these two examples there won't be this problem because you have the initialization of sum outside. If it is a matter of good practice I am not familiar with which one is preferrable...